### PR TITLE
Fix waiting for agent policy reassignment

### PR DESCRIPTION
**Pull request by nchaulet**
My take on #127 

The `agent-status` is not really reliable to know successfully get a config, this PR change it to use the `policy_revision` that is set when an agent acknowledged a `POLICY_CHANGE`

### CODE REVIEW

**Comment by ycombinator** [comment]
Thanks for the PR @nchaulet! I'm going to re-test it a few times just to make sure things are not flaky.

File: internal/kibana/ingestmanager/client_agents.go [hunk 1]
```diff
@@ -63,32 +64,51 @@ func (c *Client) AssignPolicyToAgent(a Agent, p Policy) error {
 	return nil
 }
 
+func (c *Client) getTotalAgentForPolicy(p Policy) (int, error) {
+	kuery := url.QueryEscape(fmt.Sprintf("fleet-agents.policy_id:\"%s\"", p.ID))
+	path := fmt.Sprintf("agents?kuery=%s", kuery)
+	statusCode, respBody, err := c.get(path)
+	if err != nil {
+		return -1, errors.Wrapf(err, "could not check agent status; API status code = %d; policy ID = %s; response body = %s", statusCode, p.ID, string(respBody))
```
Comment (ycombinator) [inline internal/kibana/ingestmanager/client_agents.go:72]: Should be okay to return `0` as the first return value here, yes?

Comment (ycombinator) [reply internal/kibana/ingestmanager/client_agents.go:72]: Generally speaking the Golang idiom is to return a data type's zero-value as the return value if there is an error also being returned. The expectation is that the caller will check for the error and ignore any other return values.

File: internal/kibana/ingestmanager/client_agents.go [hunk 2]
```diff
@@ -63,32 +64,51 @@ func (c *Client) AssignPolicyToAgent(a Agent, p Policy) error {
...
+			Total int `json:"total"`
 		}
 
 		if err := json.Unmarshal(respBody, &resp); err != nil {
-			return errors.Wrap(err, "could not convert agent status (response) to JSON")
-		}
-
-		if resp.Results.Total == 0 {
-			return fmt.Errorf("no agent is available")
+			return errors.Wrap(err, "could not convert agent lost (response) to JSON")
```
Comment (ycombinator) [inline internal/kibana/ingestmanager/client_agents.go:108]: Did you mean `last` instead of `lost`?

Comment (nchaulet) [reply internal/kibana/ingestmanager/client_agents.go:108]: `list` I will fix it

**Comment by ycombinator** [comment]
Going to test for flakiness one final time but this is looking pretty good!

**Review by ycombinator — APPROVED** [review]
LGTM. Thanks, again, for the fix, @nchaulet!

**Pull request merged**